src/node/connection_types.h

```diff
@@ -76,6 +77,9 @@ enum class ConnectionType {
     ADDR_FETCH,
 };
 
+/** Number of entries in ConnectionType. */
+static constexpr size_t NUM_CONNECTION_TYPES{6};
```

Comment:
Is there a way we can programmatically enforce this, like how it's done for `NUM_NET_MESSAGE_TYPES`?